### PR TITLE
chore: Allow GET requests from omniauth

### PR DESCRIPTION
**Pull request by ansor4**
So this works! I removed the lines from volt's `initializers/omniauth.rb` file and stuck them in here. 

I think next steps would be to upgrade this gem in kinetic and artsy-auth, then upgrade those gems in affected repos.

### CODE REVIEW

**Comment by starsirius** [comment]
I think we can bump the version here (and maybe even mark it as a security update so dependabot will submit PRs automatically in client repos). We probably don't need to update kinetic or artsy-auth unless there is version requirement mismatch.

**Review by joeyAghion — APPROVED** [review]
👍 from me.

This gem will need a version bump and release.

> I think next steps would be to upgrade this gem in kinetic and artsy-auth, then upgrade those gems in affected repos.

Technically, since [kinetic's](https://github.com/artsy/kinetic/blob/a325db11f4d5940c14b585dbe691571698ae4129/kinetic.gemspec#L19) and [artsy-auth's](https://github.com/artsy/artsy-auth/blob/c643e8b3110a3b9d3c8ec0bd6492bcc240c60334/artsy-auth.gemspec#L19) version specifications are flexible, you could leave them alone and just upgrade the consuming apps directly to get this additional feature:

```bash
bundle update omniauth-artsy
```

If the `allowed_request_methods` and `silence_get_warning` setters are only available on certain omniauth versions, this gem's `.gemspec` file should be updated with that constraint.

**Comment by joeyAghion** [comment]
On 2nd thought, you could avoid thorny version constraints by just leaving this to consuming apps:

```ruby
OmniAuth.config.allowed_request_methods = %i[post get] if OmniAuth.config.respond_to?(:allowed_request_methods=)
OmniAuth.config.silence_get_warning = true if OmniAuth.config.respond_to?(:silence_get_warning=)
```

**Comment by ansor4** [comment]
@starsirius having a hard time figuring out how to mark this version as a security update for dependabot, could you point me in the right direction?

**Comment by starsirius** [comment]
I haven't done this before but _I think_ we will need to PR to [ruby-advisory-db](https://github.com/rubysec/ruby-advisory-db) (which is where dependabot is sourcing from) for omniauth-artsy. I'm happy to pair on this and try it out!

File: lib/omniauth-artsy/version.rb [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 module Omniauth
   module Artsy
-    VERSION = '0.2.3'
+    VERSION = '0.3.3'
```
Comment (starsirius) [inline lib/omniauth-artsy/version.rb:5]: Maybe bumping this up to `'0.3.0'` makes more sense for a minor version upgrade?

**Pull request merged**